scm-dao-xml/src/main/java/sonia/scm/repository/xml/XmlRepositoryDAO.java

```diff
@@ -149,11 +170,20 @@ public void modify(Repository repository) {
 
     synchronized (this) {
```

Comment:
Do we still have to synchronize? Is the lock not enough?

Reply:
Rewrote synchronization once more.